hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -137,23 +153,35 @@ public void doZeroSum(final List<BalanceChange> changes) {
         if (validity == OK) {
```

Comment:
Before going into this block, we'll need something like,
```
if (validity == OK && autoCreationFee > 0) {
    if (autoCreationFee > (long) accountsLedger.get(txnCtx.activePayer(), BALANCE)) {
      validity = INSUFFICIENT_PAYER_BALANCE;
    }
}
```
Otherwise looks we'd get `FAIL_INVALID` (or else un-rolled back token changes if fee charging throws an `InvalidTransactionException`...)

Reply:
Fixed !